cmake/modules/FindDeprecated.cmake

```diff
@@ -68,6 +68,19 @@ if("CROSS_COMPILE" IN_LIST Deprecated_FIND_COMPONENTS)
   endif()
 endif()
 
+if("XTOOLS" IN_LIST Deprecated_FIND_COMPONENTS)
+  list(REMOVE_ITEM Deprecated_FIND_COMPONENTS XTOOLS)
+  # This code was deprecated after Zephyr v3.2.0
+  if(NOT DEFINED ZEPHYR_TOOLCHAIN_VARIANT)
+    set(ZEPHYR_TOOLCHAIN_VARIANT $ENV{ZEPHYR_TOOLCHAIN_VARIANT})
+  endif()
```

Comment:
because `ZEPHYR_TOOLCHAIN_VARIANT` is referenced below.

`XCC_USE_CLANG` and `CROSS_COMPILE` cases above do the same thing so I assumed `ZEPHYR_TOOLCHAIN_VARIANT` is not available at this point, is that not the case?

Reply:
done.

now that `find_package` for `XTOOLS` has been moved to `zephyr/cmake/toolchain/xtools/generic.cmake`, the `ZEPHYR_TOOLCHAIN_VARIANT` check is no longer necessary and `ZEPHYR_TOOLCHAIN_VARIANT` is indeed unnecessary.